src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -1506,17 +1506,25 @@ public void assertXPathResultsContainText(DomNode page, String xpath, String nee
         assertThat("needle found in haystack", found, is(true));
     }
 
+
+    public void assertImageLoadsSuccessfully(HtmlImage img) {
+        try {
+            assertEquals("Failed to load " + img.getSrcAttribute(),
```

Comment:
This should not happen in a helper method. This should say wether the image is loaded or not and the test will decide if that is an assertion error, which kind, etc.

Reply:
I extracted it not as a helper method but as a custom assertion on it's own for downloading single images, not all within a page.
If that's not OK I'd rather just inline the changes on the other function, as it was previously

Reply:
I'm not sure. In my opinion the test should be the one deciding what happens if the behaviour is not the expected one. For example something as simple as changing the logged message when the assertion fails.

Reply:
if you see the rest of the class, it's the way to proceed. There are plenty of methods `assertXYZ` with `assertTrue/That/Equals` within.

Reply:
unless you have a need for a new API I would probably inline it or make this method private,

the assert part of the method means that an assertion / other exception is expected to be thrown if the condition is not met

Reply:
I have inlined the assertion code into the old methods

Reply:
Resolving this one. Please open if you would like further changes